beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -240,6 +240,20 @@ func (vs *Server) unblindBuilderBlock(ctx context.Context, b interfaces.SignedBe
 	if err != nil {
 		return nil, err
 	}
+	headerRoot, err := header.HashTreeRoot()
+	if err != nil {
+		return nil, err
+	}
+
+	payloadRoot, err := payload.HashTreeRoot()
+	if err != nil {
+		return nil, err
+	}
+	if headerRoot != payloadRoot {
+		return nil, fmt.Errorf("header and payload root do not match, consider disconnect from relay to avoid further issues, "+
```

Comment:
can we just disconnect here? it seems like an attack is more likely than a bug causing this.

Reply:
That's best but hard to implement given the time. Disconnecting from the RPC level will require more thought and design. We should do this shortly post v3. Keep in mind we also have circuit breaker so prevent liveness goes down

Reply:
Fair enough